backend/src/helpers/secretScanning.ts

```diff
@@ -0,0 +1,437 @@
+      rootEncryptionKey,
+      gitSecretBlindIndexData.saltIV,
+      gitSecretBlindIndexData.saltTag
+    )
+    return salt;
+  } else if (encryptionKey && gitSecretBlindIndexData.keyEncoding === ENCODING_SCHEME_UTF8) {
+    const salt = decryptSymmetric128BitHexKeyUTF8({
+      ciphertext: gitSecretBlindIndexData.encryptedSaltCiphertext,
+      iv: gitSecretBlindIndexData.saltIV,
+      tag: gitSecretBlindIndexData.saltTag,
+      key: encryptionKey
+    })
+    return salt;
+  } else throw GitSecretBlindIndexSaltNotFoundError();
+};
+
+/**
+ * Helper function to encrypt the Git secret values [gitSecrets] provided, ignoring duplicates.
+ * @param {Object} obj - The input parameters for creating Git secret blind index data.
+ * @param {String[]} obj.gitSecrets - raw Git secret values from the Infisical Radar scan
+ * @returns {Promise<IGitSecret[]>} - A Promise that resolves to the encrypted Git secrets.
+ */
+const encryptGitSecretsHelper = async ({ gitSecrets }: EncryptGitSecretsParams): Promise<IGitSecret[]> => {
+  const encryptionKey: string = await getEncryptionKey();
+  const rootEncryptionKey: string = await getRootEncryptionKey();
+
+  const uniqueGitSecrets = new Set(gitSecrets);
+  const encryptedGitSecrets: IGitSecret[] = [];
+
+  for (const gitSecret of uniqueGitSecrets) {
+    if (rootEncryptionKey) {
+      const { ciphertext, iv, tag } = client.encryptSymmetric(gitSecret, rootEncryptionKey);
+
+      const newGitSecret = new GitSecret({
+        gitSecretValueCiphertext: ciphertext,
+        gitSecretValueIV: iv,
+        gitSecretValueTag: tag,
+        algorithm: ALGORITHM_AES_256_GCM,
+        keyEncoding: ENCODING_SCHEME_BASE64,
+      })
+
+      encryptedGitSecrets.push(newGitSecret);
+    } else {
+      const { ciphertext, iv, tag } = encryptSymmetric128BitHexKeyUTF8({
+        plaintext: gitSecret,
+        key: encryptionKey,
+      })
+
+      const newGitSecret = new GitSecret({
+        gitSecretValueCiphertext: ciphertext,
+        gitSecretValueIV: iv,
+        gitSecretValueTag: tag,
+        algorithm: ALGORITHM_AES_256_GCM,
+        keyEncoding: ENCODING_SCHEME_UTF8,
+      })
+
+      encryptedGitSecrets.push(newGitSecret);
+    }
+  }
+
+  return encryptedGitSecrets;
+};
+
+/**
+ * Create secret blind index data containing encrypted blind index salt
+ * for the GitHub organization's ID [organizationId].
+ * @param {Object} obj - The input parameters for creating Git secret blind index data.
+ * @param {String} obj.organizationId - GitHub organization's ID for Infisical Radar.
+ * @returns {Promise<string>} - A Promise that resolves to the generated Git secret blind index salt for the organization.
+ */
+export const createGitSecretBlindIndexDataHelper = async ({
+  organizationId
+}: CreateGitSecretBlindIndexDataParams): Promise<string> => { 
+  // initialize random blind index salt for the organization
+  const randomBytesPromisified = promisify(randomBytes);
```

Comment:
what is promisify and why do we need it? I don't see any instances were we were using it before